nixos/modules/services/web-apps/shiny-server.nix

```diff
@@ -0,0 +1,129 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.shiny-server;
+
+  generatedConfig = pkgs.writeText "shiny-server.conf" ''
+    run_as ${cfg.user};
+    server {
+      listen ${toString cfg.port};
+      location / {
+        site_dir ${cfg.siteDir};
+        log_dir ${cfg.logDir};
+        bookmark_state_dir ${cfg.stateDir};
+        directory_index ${boolToString cfg.directoryIndex};
+      }
+    }
+    ${cfg.extraConfig}
+  '';
+in
+{
+  options = {
+    services.shiny-server = {
+      enable = mkEnableOption "Enable server to host R Shiny applications.";
+
+      package = mkOption {
+        default = pkgs.shiny-server.override {
+          extraRPackages = cfg.extraPackages;
+        };
+        defaultText = ''pkgs.shiny-server.override {
+          extraRPackages = cfg.extraPackages;
+        };
+        '';
+        description = "Shiny Server package";
+      };
+
+      extraPackages = mkOption {
+        default = [];
+        example = ''with pkgs.rPackages; [ ggplot2 data_table ];'';
+        description = ''
+          Extra packages to include in the rWrapper used by shiny-server.
+          Use this to enable additional R packages.
+        '';
+      };
+
+      port = mkOption {
+        type = types.int;
+        default = 3838;
+        description = ''
+          Port number Shiny Server will be listening to.
+        '';
+      };
+
+      directoryIndex = mkOption {
+        type = types.bool;
+        default = true;
+        description = "Whether directory content is displayed";
+      };
+
+      user = mkOption {
+        default = "shiny";
+        description = "User to run Shiny Server as";
+      };
+
+      siteDir = mkOption {
+        type = types.path;
+        default = "${cfg.package}/shiny-server/samples";
+        defaultText = ''''${cfg.package}/shiny-server/samples'';
```

Comment:
let's just use `"`s here instead of the unneeded `''`s.

Reply:
Done.